src/storage-controller/src/healthcheck.rs

```diff
@@ -0,0 +1,467 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::{BTreeMap, BTreeSet};
+use std::sync::Arc;
+
+use chrono::{DateTime, NaiveDateTime, Utc};
+use differential_dataflow::lattice::Lattice;
+use mz_ore::now::EpochMillis;
+use mz_persist_types::Codec64;
+use mz_repr::{Datum, Diff, GlobalId, Row, TimestampManipulation};
+use timely::progress::Timestamp;
+
+use crate::collection_mgmt::CollectionManager;
+use crate::IntrospectionType;
+
+pub use mz_storage_client::healthcheck::*;
+
+pub fn pack_status_row(
```

Comment:
Since we already have a struct for these arguments it would nicer to have it be `fn pack_status_row(status: &RawStatusUpdate) -> Row`. It will also save you a bunch of destructuring in other places